File: Assignment 2/TCPServer.go
/**
 * TCPServer.go
 **/

package main

import ("bytes"; "fmt"; "net")

func main() {
    serverPort := "12000"

    listener, _:= net.Listen("tcp", ":" + serverPort)
    fmt.Printf("Server is ready to receive on port %s\n", serverPort)

    buffer := make([]byte, 1024)

    for {
        conn, _:= listener.Accept()
        fmt.Printf("Connection request from %s\n", conn.RemoteAddr().String())
        count, _ := conn.Read(buffer)
        conn.Write(bytes.ToUpper(buffer[:count]))
        conn.Close()
    }
}

File: Assignment 2/TCPServer.py
#
# SimpleEchoTCPServer.py
#

from socket import *

serverPort = 12000
serverSocket = socket(AF_INET, SOCK_STREAM)
serverSocket.bind(('', serverPort))
serverSocket.listen(1)

print("Server is ready to receive on port", serverPort)

while True:
    (connectionSocket, clientAddress) = serverSocket.accept()
    print('Connection request from', clientAddress)
    message = connectionSocket.recv(2048)
    modifiedMessage = message.decode().upper()
    connectionSocket.send(modifiedMessage.encode())
    connectionSocket.close()

